### ISSUE TITLE
CA1848 false positive on `LoggerMessage` methods

### ISSUE THREAD

**Issue by TaffarelJr** [dotnet/roslyn-analyzers#7150]
### Analyzer

**Diagnostic ID**: [CA1848](https://learn.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1848): `Use the LoggerMessage delegates`

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://learn.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 8.0.101](https://dotnet.microsoft.com/download/dotnet/8.0)

### Describe the bug

After implementing the [LoggerMessage](https://learn.microsoft.com/en-us/dotnet/core/extensions/high-performance-logging) pattern as prescribed, this analyzer appears to mistake my strongly-typed extension methods for the built-in legacy extension methods (`.LogTrace`, `.LogInformation`, etc.) and triggers false positives on every single instance.

### Steps To Reproduce

Define high-performance log delegate, along with its matching strongly-typed extension method:

```csharp
private static readonly Action<ILogger, string, int, int, Exception?> _gettingAllChats =
    LoggerMessage.Define<string, int, int>(
        LogLevel.Trace,
        new EventId((int)LoggerEventId.GettingAllChats, nameof(GettingAllChats)),
        "Retrieving all chats for user {UserId} (page {PageNumber}, size {PageSize})");

public static void GettingAllChats(
    this ILogger logger,
    string userId,
    int pageNumber,
    int pageSize)
{
    _gettingAllChats(logger, userId, pageNumber, pageSize, null);
}
```

Call the extension method from code, elsewhere:

```csharp
// Set during constructor. Optional.
// May be null if logging is not configured in the app.
private readonly ILogger? _logger = logger;

...

_logger?.GettingAllChats(userId, pageNumber, pageSize);
```

### Expected behavior

This code was put in place specifically to satify this Analyzer (replaces legacy `.LogTrace` calls), so we would expect the warnings to go away.

### Actual behavior

The analyzer logs a false positive, with the following content:

| Severity | Code | Description | Suppression State |
|:---:|:---:|-----|:---:|
| Warning | CA1848 | For improved performance, use the LoggerMessage delegates instead of calling 'LoggerExtensions.GettingAllChats(ILogger, string, int, int)' | Active |

## Additional context

I'm using an enum for the EventIDs in our delegates, so they automatically increment. Doesn't seem to be part of triggering the false positive though, because when I replace the value with a standard `int` I still get the warnings.

**Comment by mavasani** [comment]
@maryamariyan

**Comment by TaffarelJr** [comment]
Refactoring the extension method to use the new `LoggerMessage` attribute does not resolve it either:

```csharp
[LoggerMessage(
    (int)LoggerEventId.GettingAllChats,
    LogLevel.Trace,
    "Retrieving all chats for user {UserId} (page {PageNumber}, size {PageSize})")]
public static partial void GettingAllChats(
    this ILogger logger,
    string userId,
    int pageNumber,
    int pageSize);
```
